include/verilated_random.h

```diff
@@ -510,9 +518,32 @@ class VlRandomizer final {
         }
     }
 
-    // TODO: Add support for associative arrays of structs
     // Recursively process associative arrays of structs
+    template <typename T_Key, typename T_Value>
+    void record_struct_arr(VlAssocArray<T_Key, T_Value>& var, const std::string name,
+                           int dimension, std::vector<IData> indices,
+                           std::vector<size_t> idxWidths) {
+        if ((dimension > 0) && (var.size() != 0)) {
```

Comment:
```suggestion
        if ((dimension > 0) && !var.empty()) {
```

(may need to add empty() method).